File: src/difficulty_1/difficulty_1_3/lastFactorialDigit/LastFactorialDigit.java
package difficulty_1.difficulty_1_3.lastFactorialDigit;

import java.util.Scanner;

public class LastFactorialDigit {

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        int nrOfTests = scanner.nextInt();

        for (int i = 0; i < nrOfTests; i++) {
            int nr = scanner.nextInt();
            int result = 1;

            for (int factor = 2; factor <= nr; factor++) {
                result *= factor;
                result %= 10;
            }

            System.out.println(result);
        }
    }
}

File: KotlinImplementation/src/difficulty_1/LastFactorialDigit.kt
package difficulty_1

fun main() {
    val n = readLine()!!.toInt()

    for(case in 1..n) {
        val nr = readLine()!!.toInt()

        var result = 1

        for(factor in 2..nr) {
            result *= factor
            result %= 10
        }

        println(result)
    }
}